Science/src/temp_humidity/temp_humidity.cpp

```diff
@@ -1,23 +1,57 @@
 #include "temp_humidity.h"
 
-TempHumiditySensor::TempHumiditySensor() : sensor(&Wire, 0x44, 4) { }
+void TempHumiditySensor::readReg(){
+  Wire.beginTransmission(addr);
+  // Rewrote with overloaded method so that you don't need to use & on a literal
+  Wire.write(reg);
+  uint8_t code = Wire.endTransmission();
+  if(code != 0){
+    // Wiping buffer if there is an error
+    memset(buf, 0, 4);
+    Serial.print("Error: Could not read temperature/humidity (Error code ");
+    Serial.print(code);
+    Serial.println(")");
+    return;
+  }
+
+  delay(20);
+  
+  // Requesting 4 bytes to get data from registers 0x00 and 0x01 and putting it in buf[0:4]
+  Wire.requestFrom(addr, 4);
+  for(int i = 0; i < 4; i++){
+    buf[i] = Wire.read();
+  }
+  return;
+}
+
+void TempHumiditySensor::setup(){ 
+  // 5 second delay before issuing error
+  unsigned long startTime = millis();
+  unsigned long timeout = 5000;
 
-void TempHumiditySensor::setup() {
   Serial.print("Initializing temp/humidity sensor...");
-  while (sensor.begin() != 0) {
+  while (Wire.begin() != 1) {
     Serial.print(".");
+     if (millis() - startTime > timeout) {
+        // Timeout reached, so issue an error
+        Serial.println("Sensor setup failed.");
+        return;
+    }
     delay(500);
   }
   Serial.println(" Done!");
-  if (!sensor.softReset()) {
-    Serial.println("Failed to Initialize the chip....");
-  }
 }
 
-float TempHumiditySensor::getTemperature() {
-  return sensor.getTemperatureC();
+float TempHumiditySensor::getTemperature(){
+  readReg();
+  uint16_t temp = buf[0] << 8 | buf[1];
+  return ((float) temp * 165 / 65535.0) - 40.0;
```

Comment:
Definitely a style choice, but consider this: `getTemperature()` and `getHumidity()` are probably going to be called in the same `loop()` function, and so `readReg()` will be run twice in a row. Ignoring the delays, it's unlikely that the sensor itself will have new data in such a short time. 

Our other classes have a `void update()` method -- something meant to be run once per `loop()`. Maybe you can put `readReg()` in there, and save the temperature and humidity into their own fields. Then, when these functions are called, just return the last read value from the fields directly. 

It's a bit of a trade-off from efficiency vs real-time accuracy, so your call.